src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/EventListener.java

```diff
@@ -164,25 +166,45 @@ protected void schedule(GerritTrigger t, GerritCause cause, GerritTriggeredEvent
      * @param event   the event.
      * @param project the project to build.
      */
-    protected void schedule(GerritTrigger t, GerritCause cause, GerritTriggeredEvent event, AbstractProject project) {
+    protected void schedule(GerritTrigger t, GerritCause cause, GerritTriggeredEvent event, Job project) {
         BadgeAction badgeAction = new BadgeAction(event);
         //during low traffic we still don't want to spam Gerrit, 3 is a nice number, isn't it?
         int projectbuildDelay = t.getBuildScheduleDelay();
         if (cause instanceof GerritUserCause) {
             // it's a manual trigger, no need for a quiet period
             projectbuildDelay = 0;
-        } else if (project.getHasCustomQuietPeriod()
-                && project.getQuietPeriod() > projectbuildDelay) {
-            projectbuildDelay = project.getQuietPeriod();
+        } else if (project instanceof AbstractProject) {
+            // TODO: How about using ParameterizedJobMixIn.ParameterizedJob here?
```

Comment:
Agreed. The only other possibility would be if the user pushes several separate commits using separate git-push...

I think that's rather not worth it to care much about it, but the plugin already does support things this way...

I don't really get the idea here, but i think it's something like "if job has its own quiet period, use that, unless we set a higher build delay ourselves, then use that instead.  But if a project doesn't have a customized build delay and only has the global quiet period, just use our value instead of the larger of the values.

That seems really convoluted logic, and we could just use "max" of quiet period or build delay.

I think the primary reason we have build delay is incase of git mirrors, where you clone from a mirror, but monitor based on a gerrit server, and want to wait long enough for the mirror to be updated so that it has the refs you need to pull.